packages/vite/src/node/plugins/asset.ts

```diff
@@ -199,11 +199,13 @@ async function fileToBuiltUrl(
   let url
   if (
     config.build.lib ||
-    (!file.endsWith('.svg') &&
-      content.length < Number(config.build.assetsInlineLimit))
+    Buffer.byteLength(content) < config.build.assetsInlineLimit
   ) {
-    // base64 inlined as a string
-    url = `data:${mime.getType(file)};base64,${content.toString('base64')}`
+    // svgs can be inlined without base64
+    url = file.endsWith('.svg')
```

Comment:
While `/\.svg$/i.test(file)` is more fool-proof, I don't think all-caps file extensions are particularly common?

Reply:
I used plain `endsWith` merely to be consistent with the rest of the codebase. If someone ever decides to supports case-insensitive extensions it'll be an easy fix.

Reply:
> I don't think all-caps file extensions are particularly common?

It shouldn't be common no. Depends if you want to avoid it ever being an issue for users that do though.

Not a big deal, worse that happens is it gets encoded as base64 instead and if that bothers/confuses a user, they raise an issue and someone can point out lowercase extension is only supported or someone adds the fix by lowercasing the filename (only for checking the extension).

> If someone ever decides to supports case-insensitive extensions it'll be an easy fix.

I don't know what you mean here? You just use [`toLowerCase()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLowerCase) on the string.

```js
url = file.toLowerCase().endsWith('.svg')
```